src/components/raycaster.js

```diff
@@ -31,6 +31,15 @@ module.exports.Component = registerComponent('raycaster', {
     this.prevCheckTime = undefined;
     this.raycaster = new THREE.Raycaster();
     this.updateOriginDirection();
+    this.refreshObjects = this.refreshObjects.bind(this);
+  },
+
+  play: function () {
+    this.el.sceneEl.addEventListener('child-attached', this.refreshObjects);
```

Comment:
I don't think we should compromise on performance. Imagine a shooting game where the gun is a raycaster aimed at monsters, and you are spawning lots of bullets. This would rebuild the entire raycaster target list on each bullet when we actually don't have to do anything. You don't have to worry about children because `child-attached` will be called on each child and the raycaster can handle recursion.